crates/services/txpool_v2/src/pending_pool.rs

```diff
@@ -229,16 +229,13 @@ impl PendingPool {
     }
 
     // We expect it to be called a lot (every new tx in the pool).
-    pub fn new_known_tx(
+    pub fn new_known_tx<'a>(
         &mut self,
-        new_known_tx: ArcPoolTx,
+        new_known_tx_outputs: impl Iterator<Item = &'a Output>,
```

Comment:
perhaps this `Output` should be boxed? 
difficult to say without knowing average size of outputs that we have, but in the case the user has 255 outputs this would be overusing the cache and move other imp code out of the cpu cache

since we have a limit on the total number of outputs, perhaps we can use smallvec?

Reply:
(i only write this because of the comment that says "we expect it to be called a lot")